### ISSUE TITLE
Generator not working on functions that accepts type arguments (TypeScript)

### ISSUE THREAD

**Issue by filipekiss** [kkoomen/vim-doge#79]
First of all, DoGe is great and it works perfectly 99% of the time. Thank you so much for your hard work!

Generally I don't have an issue with TypeScript, except when I'm writing functions that accept type arguments:

```typescript
export function rejectMapByValue<MapItem = unknown, MapKey = unknown>(
  x: (xs: MapItem) => Boolean,
): (xs: Map<MapKey, MapItem>) => MapItem[] {
  return xs => {
    return [...xs.values()].filter(xse => {
      return !applyFilter<MapItem>(x)(xse);
    });
  };
}
```

When this is the case, the generator doesn't work at all. If I remove the type arguments, it works as expected:

```typescript
/**
 * [TODO:description]
 *
 * @param {[TODO:type]} x - [TODO:description]
 * @return {[TODO:type]} [TODO:description]
 */
export function rejectMapByValue(x) {
  return xs => {
    return [...xs.values()].filter(xse => {
      return !applyFilter<MapItem>(x)(xse);
    });
  };
}
```
**Settings**
```viml
let g:doge_comment_jump_modes = ['n', 's']
```
**To Reproduce**
1. Create any TypeScript file (`.ts` or `.tsx`)
1. Declare a function with type arguments (or use the one at the beggining of the issue)
1. Try to generate documentation for it
1. Nothing happens…
1. Remove the type arguments and try again. Now it works as expected

**Expected behavior**

For now, simply ignoring the type arguments when parsing would be enough and allow the generator to keep using JSDoc. I assume this will be handled by a future TSDoc spec, so I don't see a point on trying to generate compatible JSDoc for this.

**Out of `vim --version`**
nvim v0.4.3

**Comment by kkoomen** [comment]
Hi. Thanks for your compliment about DoGe. It has been appreciated.

I pushed a fix, including passing tests, to the master branch. Can you let me know if you come across any other scenarios? I've fixed your scenario and kept using jsdoc standards.

This is the result I get locally now:
```javascript
/**
 * [TODO:description]
 *
 * @param {(xs: MapItem) => Boolean} x - [TODO:description]
 * @param {(xs: MapItem) => Boolean} y - [TODO:description]
 * @return {[TODO:type]} [TODO:description]
 */
export function rejectMapByValue<MapItem = unknown, MapKey = unknown>(
  x: (xs: MapItem) => Boolean,
  y: (xs: MapItem) => Boolean,
): (xs: Map<MapKey, MapItem>) => MapItem[] {}
```

although I am not sure if ` * @param {(xs: MapItem) => Boolean} x - [TODO:description]` should be ` * @param {(xs: MapItem): Boolean} x - [TODO:description]` or not (the `=>` is changed into `:`. Do you perhaps know this?

**Comment by filipekiss** [comment]
I think keeping the `=>` is ok, since that's how we type the function signatures, usually. I just tested and everything seems to be working as intended. Thanks for your quick response!

**Comment by kkoomen** [comment]
Awesome. I will release a new version immediately.

**Comment by filipekiss** [comment]
Thank you so much :D I'll close the issue since it's resolved!

**Comment by kkoomen** [comment]
This bug has been fixed and released in [v2.2.2](https://github.com/kkoomen/vim-doge/tree/v2.2.2).

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.
